training/model_management/components/import_model/spec.yaml

```diff
@@ -0,0 +1,135 @@
+$schema: https://azuremlschemas.azureedge.net/latest/pipelineComponent.schema.json
+type: pipeline
+
+name: import_model
+display_name: Import model
+description: Import a model into a workspace or a registry
+version: 0.0.1
+
+# Pipeline inputs
+inputs:
+  # pipeline specific compute
+  compute:
+    type: string
+    description: Compute to run pipeline job
+
+  ## Inputs for download model
+  model_source:
+    type: string
+    description: Storage containers from where model will be sourced from.
+    default: Huggingface
+    enum:
+      - AzureBlob
+      - GIT
+      - Huggingface
+
+  model_id:
+    type: string
+    description: A valid model id for the model source selected. For example you can specify `bert-base-uncased` for importing HuggingFace bert base uncased model. Please specify the complete URL if **GIT** or **AzureBlob** is selected in `model_source`
+
+  ## Inputs for the MlFLow conversion
+  license_file_path:
+    type: uri_file
+    description: Path to the license file
+    optional: true
+
+  ## Inputs for model local validation
+  test_data_path:
+    type: uri_file
+    optional: true
+    description: Test dataset for model inferencing
+
+  test_data_column_rename_map:
+    type: string
+    optional: true
+    description: |
+      Provide mapping of dataset column names that should be renamed before inferencing.
+      eg: col1:ren1; col2:ren2; col3:ren3
+
+  ## Inputs for Model registration
+  custom_model_name:
+    type: string
+    optional: true
+    description: Model name to use in the registration. If name already exists, the version will be auto incremented
+
+  model_version:
+    type: string
+    optional: true
+    description: Model version in workspace/registry. If the same model name and version exists, the version will be auto incremented
+
+  model_description:
+    type: string
+    optional: true
+    description: Description of the model that will be shown in AzureML registry or workspace
+
+  registry_name:
+    type: string
+    optional: true
+    description: Name of the AzureML asset registry where the model will be registered. Model will be registered in a workspace if this is unspecified
+
+  model_metadata:
+    type: uri_file
+    optional: true
+    description: A JSON or a YAML file that contains model metadata confirming to Model V2 [contract](https://azuremlschemas.azureedge.net/latest/model.schema.json)
```

Comment:
How do we make this simpler?

Reply:
name, version and description are present as simpler inputs. For complex types such as model flavor, tags, properties YAML or JSON file adhering to v2 contract can be passed.

tags and properties are (key, value) string pairs. For them we can accept string like:
` a:b; c:d; e:f .... `

This can be added in registration component and supported accordingly. Flavor though can contain complex value for keys as well. eg: 

```
flavors:
  hftransformers:
    code: null
    hf_config_class: GPT2Config
    hf_pretrained_class: AutoModelWithLMHead
    hf_tokenizer_class: GPT2TokenizerFast
    huggingface_id: gpt2-large
    model_data: data
    pytorch_version: 2.0.0+cu117
    task_type: text-generation
    transformers_version: 4.26.1
  python_function:
    data: data
    env: conda.yaml
    loader_module: azureml.evaluate.mlflow.hftransformers
    python_version: 3.8.16
```

Reply:
Let us take this up in the next iteration. 

Cc: @ManojBableshwar @swatig007